HeterogeneousCore/CUDAUtilities/src/CachingDeviceAllocator.h

```diff
@@ -299,7 +299,7 @@ namespace notcub {
      */
     cudaError_t SetMaxCachedBytes(size_t max_cached_bytes) {
       // Lock
-      mutex.lock();
+      std::unique_lock mutex_locker(mutex);
```

Comment:
Please add a comment long
```suggestion
      // CMS: use RAII instead of (un)locking explicitly
      std::unique_lock mutex_locker(mutex);
```